packages/core/src/codewhisperer/service/transformByQ/transformApiHandler.ts

```diff
@@ -385,40 +385,182 @@ export function getImageAsBase64(filePath: string) {
     return `data:image/svg+xml;base64,${fileContents}`
 }
 
+/*
+ * Given the icon name from core/resources/icons/aws/amazonq, get the appropriate icon according to the user's theme.
+ * ex. getIcon('transform-file') returns the 'transform-file-light.svg' icon if user has a light theme enabled,
+ * otherwise 'transform-file-dark.svg' is returned.
+ */
+export function getIcon(iconName: string) {
```

Comment:
Maybe rename this to `getAmazonQIcon()` since we already have `getIcon()`

Reply:
why doesn't this use the existing getIcon util ?

Reply:
> so are light/dark versions of the icon somehow generated for you and if so how?

yes. explained in:

- https://github.com/aws/aws-toolkit-vscode/blob/master/CONTRIBUTING.md#font-generation
- https://github.com/aws/aws-toolkit-vscode/blob/master/docs/icons.md , which is linked from

Reply:
Just spent some time looking into this. For some reason when I add my icons under `resources/icons/aws/amazonq/dark/` and `resources/icons/aws/amazonq/light/` (following instructions here: https://github.com/aws/aws-toolkit-vscode/blob/master/docs/icons.md#theme-overrides) then run `npm run generatePackage` the `package.json` is not updated with the icons I just added. Some changes appear to be made but my new icons are not added. When I move the `dark/` and `light/` folders to be directly under `aws/` that also does not work. I have to add the icons directly under the existing `amazonq/` folder for `npm run generatePackage` to update the `package.json` properly. I also noticed that other folders (`applicationcomposer/`, `codewhisperer/`, and `mynah/`) simply put a dark and light version of the icon directly in their folder without a `dark/` or `light/` sub-folder, which matches my original implementation in this PR.

Not sure what I'm doing wrong. I would like to follow up on this but that will have to be post-GA because this PR likely needs to be merged by EOD tomorrow and I still have a couple of other small changes to make, so I'll make a SIM for this.

Reply:
SIM ID is EGCP-130